fix_checkpoint_history.py

```
@@ -0,0 +1,80 @@
+    checkpoint_file = Path("data/openaq/checkpoints/checkpoint_jp_all_parallel.json")
+    checkpoint_data = {
+        "country_code": "JP",
+        "location_index": estimated_locations,
+        "total_locations": 1607,
+        "completed_locations": [],  # Empty for now, will be filled as processing continues
+        "output_file": target_csv,
+        "current_location_id": None,
+        "timestamp": datetime.now().isoformat()
+    }
+    
+    with open(checkpoint_file, 'w') as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)